### ISSUE TITLE
Support styled(Component)(({ theme }) => `

### ISSUE THREAD

**Issue by oliviertassinari** [hudochenkov/postcss-styled-syntax#31]
I think it would be great to support this pattern `styled(Component)(({ theme }) => ` . Today, it doesn't work:

https://github.com/hudochenkov/postcss-styled-syntax/assets/3165635/bd24c10d-8f8f-4012-8e76-7b4c71f0f61a

This is pretty much the same request as in https://github.com/styled-components/vscode-styled-components/issues/194.

**Comment by hudochenkov** [comment]
There could be more complexity hidden, that an example. Function could return template literals conditionally, or assign it to a variable, or use explicit return.

```js
const StyledDiv = styled.div((props) => {
	if (props.isDisabled) {
		return `
			background-color: #f9f9f9;
		`;
	}

	return `
		background-color: blue;
	`;
});

const StyledDiv2 = styled.div((props) => {
	let styles = `
		background-color: blue;
	`;

	if (props.isDisabled) {
		styles = `
			background-color: #f9f9f9;
		`;
	}

	return styles;
});
```

Additionally, I doubt VS Code extension would highlight examples above. Not that it would be blocker, but worth to keep in mind.

**Comment by oliviertassinari** [comment]
> Function could return template literals conditionally, or assign it to a variable, or use explicit return.

I guess it's for these use cases that we need to use the `css` helper.

> I doubt VS Code extension would highlight examples above. Not that it would be blocker, but worth to keep in mind.

I see this not correctly highlighted/autocompleted by the VS Code extension:

```jsx
const StyledDiv = styled.div((props) => {
	return `
		background-color: #f9f9f9;
	`;
});
```

but this one is correctly highlighted/autocompleted:

```jsx
const StyledDiv = styled.div((props) => `
	background-color: #f9f9f9;
`);
```

**Comment by hudochenkov** [comment]
Alright, let's support only implicit return of template literal for an arrow function. I could take a look in a few weeks. PR is always welcome :)

**Comment by hudochenkov** [comment]
Released in `0.7.0`. Let me know, if there any issues.

**Comment by oliviertassinari** [comment]
Very cool, it seems to work

<img width="1039" alt="SCR-20241105-bocb" src="https://github.com/user-attachments/assets/8e10a5f4-0b31-4b12-ae1d-c3acfc40109d">

> 411 errors

Someone at http://github.com/mui/ will have fun to fix the codebase 😁
